[#19410] wizard send collectibles

ulisesmac opened the pull request:
fixes #19410 
fixes #16990

The bridge feature will be addressed in
- #19516

### Summary

This PR uses the wizard mechanism to unify the send token and send collectible flows, making able to skip some steps for the sending of collectibles.

Video containing the regular flow and the flow of first picking a collectible (skips select asset screen):

[Screencast from 2024-04-03 13-13-04.webm](https://github.com/status-im/status-mobile/assets/90291778/4927e2ab-f7f7-4b1a-93cc-9b010feab71a)

Flow from home wallet screen (adds from step and skips sleect asset):

[Screencast from 2024-04-03 13-14-38.webm](https://github.com/status-im/status-mobile/assets/90291778/9c35b401-db7f-4f52-b814-6538e1df78b9)

Additionally, solves the following error when going back in the navigation:

[Screencast from 2024-04-03 13-02-27.webm](https://github.com/status-im/status-mobile/assets/90291778/1fa055d2-4185-4841-ac23-b9fabbe66bbb)


### Testing notes
This PR focuses on the flow, there may be some other areas that still need to be polished/adjusted.

#### Platforms

- Android
- iOS

### Steps to test

- Open Status
- Use/Recover an account owning some collectibles
- Go to the account details
- Move to Collectibles tab
- Press any collectible
- Press "Send" and complete the send flow.

status: ready

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 1b71e41962ad4b0231f395e543c0e3a8fcaca28e | [#2](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-19515/2/) | 2024-04-03 19:49:39 | ~5 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240403-194343-1b71e41962ad4b0231f395e543c0e3a8fcaca28e-pr19515-x86_64.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-240403-194343-1b71e41962ad4b0231f395e543c0e3a8fcaca28e-pr19515-x86_64.apk)|

ulisesmac commented on src/status_im/contexts/wallet/collectible/view.cljs:
```diff
@@ -96,11 +101,11 @@
         :label               (i18n/label :t/share-collectible)
         :right-icon          :i/external}]]]))
 
-(defn f-view-internal
-  [{:keys [theme] :as _props}]
+(defn view-internal
+  [_]
   (let [selected-tab  (reagent/atom :overview)
         on-tab-change #(reset! selected-tab %)]
-    (fn []
+    (fn [{:keys [theme] :as _props}]
```
An additional fix, we were receiving the `theme` in the wrong place, the component wouldn't re-render if it changed

J-Son89 replied:
can also use the hook directly, if you prefer 👍

ulisesmac commented on src/status_im/contexts/wallet/send/events.cljs:
```diff
@@ -99,27 +100,37 @@
+   (let [one-collectible? (= (collectible.utils/collectible-balance collectible) 1)
+         collectible-tx   (-> db
+                              (update-in [:wallet :ui :send] dissoc :token)
+                              (assoc-in [:wallet :ui :send :collectible] collectible)
+                              (assoc-in [:wallet :ui :send :tx-type] :collectible))]
+     {:db (cond-> collectible-tx
+            one-collectible? (assoc-in [:wallet :ui :send :amount] 1))
+      :fx [(when one-collectible?
+             [:dispatch [:wallet/get-suggested-routes {:amount 1}]])
+           [:dispatch
+            [:wallet/wizard-navigate-forward
+             {:current-screen current-screen
+              :flow-id        :wallet-flow}]]]})))
 
-(rf/reg-event-fx :wallet/select-collectibles-amount
- (fn [{:keys [db]} [{:keys [collectible stack-id]}]]
-   {:db (-> db
-            (update-in [:wallet :ui :send] dissoc :token)
-            (assoc-in [:wallet :ui :send :collectible] collectible)
-            (assoc-in [:wallet :ui :send :tx-type] :collectible))
-    :fx [[:navigate-to-within-stack [:screen/wallet.select-collectible-amount stack-id]]]}))
+(rf/reg-event-fx
+ :wallet/set-collectible-amount-to-send
+ (fn [{db :db} [{:keys [stack-id amount]}]]
+   {:db (assoc-in db [:wallet :ui :send :amount] amount)
+    :fx [[:dispatch [:wallet/get-suggested-routes {:amount amount}]]
+         [:dispatch
+          [:wallet/wizard-navigate-forward
+           {:current-screen stack-id
+            :flow-id        :wallet-flow}]]]}))
```
Redefined these events to handle the wizard flow, and changed their names.

`send-collectibles-amount` -> `set-collectible-to-send`
`select-collectibles-amount` -> `set-collectible-amount-to-send`

Given the functionality changed and the places where they are being used, the new names match better

ulisesmac commented on src/status_im/contexts/wallet/send/events.cljs:
```diff
@@ -99,27 +100,37 @@
+      :fx [(when one-collectible?
+             [:dispatch [:wallet/get-suggested-routes {:amount 1}]])
+           [:dispatch
+            [:wallet/wizard-navigate-forward
+             {:current-screen current-screen
+              :flow-id        :wallet-flow}]]]})))
 
-(rf/reg-event-fx :wallet/select-collectibles-amount
- (fn [{:keys [db]} [{:keys [collectible stack-id]}]]
-   {:db (-> db
-            (update-in [:wallet :ui :send] dissoc :token)
-            (assoc-in [:wallet :ui :send :collectible] collectible)
-            (assoc-in [:wallet :ui :send :tx-type] :collectible))
-    :fx [[:navigate-to-within-stack [:screen/wallet.select-collectible-amount stack-id]]]}))
+(rf/reg-event-fx
+ :wallet/set-collectible-amount-to-send
+ (fn [{db :db} [{:keys [stack-id amount]}]]
+   {:db (assoc-in db [:wallet :ui :send :amount] amount)
+    :fx [[:dispatch [:wallet/get-suggested-routes {:amount amount}]]
+         [:dispatch
+          [:wallet/wizard-navigate-forward
+           {:current-screen stack-id
+            :flow-id        :wallet-flow}]]]}))
 
-(rf/reg-event-fx :wallet/send-select-amount
+(rf/reg-event-fx
+ :wallet/set-token-amount-to-send
```
According to the previous mentioned renaming, and  given that this even is specific for tokens and not collectibles, I performed this rename

ulisesmac commented on src/status_im/contexts/wallet/send/select_asset/view.cljs:
```diff
@@ -94,8 +87,4 @@
          :on-change       on-change-tab}]
        [tab-view @search-text @selected-tab on-change-text]])))
 
-(defn- view-internal
-  []
-  [:f> f-view-internal])
```
Just removed this (now unnecessary) wrapper

ulisesmac commented on src/status_im/contexts/wallet/send/events.cljs:
```diff
@@ -99,27 +100,37 @@
                      dissoc
                      :collectible
                      :amount
-                     (when (= transaction-type :collecible) :tx-type))})))
+                     (when (= transaction-type :collectible) :tx-type))})))
```
This fix is not directly related, but noticed we had a typo here :eyes:

OmarBasem commented on src/status_im/contexts/wallet/send/events.cljs:
```diff
@@ -99,27 +100,37 @@
+         collectible-tx   (-> db
+                              (update-in [:wallet :ui :send] dissoc :token)
+                              (assoc-in [:wallet :ui :send :collectible] collectible)
+                              (assoc-in [:wallet :ui :send :tx-type] :collectible))]
+     {:db (cond-> collectible-tx
+            one-collectible? (assoc-in [:wallet :ui :send :amount] 1))
+      :fx [(when one-collectible?
+             [:dispatch [:wallet/get-suggested-routes {:amount 1}]])
+           [:dispatch
+            [:wallet/wizard-navigate-forward
+             {:current-screen current-screen
+              :flow-id        :wallet-flow}]]]})))
 
-(rf/reg-event-fx :wallet/select-collectibles-amount
- (fn [{:keys [db]} [{:keys [collectible stack-id]}]]
-   {:db (-> db
-            (update-in [:wallet :ui :send] dissoc :token)
-            (assoc-in [:wallet :ui :send :collectible] collectible)
-            (assoc-in [:wallet :ui :send :tx-type] :collectible))
-    :fx [[:navigate-to-within-stack [:screen/wallet.select-collectible-amount stack-id]]]}))
+(rf/reg-event-fx
+ :wallet/set-collectible-amount-to-send
```
I think we can have tests for these events

J-Son89 commented:
@ulisesmac - I tested out this branch. 
It works great for sending an asset.
One thought - there is also sending an asset from the home page by long press on a collectible.
@mmilad75 has added the "Select From" page for this flow already so it should be rather straightforward to include that. 
Shall we add it here too? 🤔

J-Son89 commented on src/status_im/contexts/wallet/send/flow_config.cljs:
```diff
@@ -1,13 +1,30 @@
+  [db]
+  (-> db :wallet :ui :send :token some?))
+
 (def steps
```
@ulisesmac - can you instead make a separate flow for sending an asset vs a collectible.
I think it will be cleaner this way. We should also do the same for bridging imo. 

There's also an issue with sending a collectible this way. 

The issue is this, if you are on the home page - long press on a collectible.
When you click send it will bring you to the "From" Page.
This From page shows all addresses 
- it's unclear if we want this, probably we want to show that page only if that collectible is owned by many accounts by the user.


Also in the case of a user do we really want to allow a user to select an account that doesn't have that asset to send? 🤔 
Seems like the other accounts should be greyed out, probably a separate issue.

ulisesmac replied:
Yeah, since while sending a token, we also display all accounts, I wasn't sure if we really wanted to filter.

Maybe better if this filtering for sending collectibles and assets is first calrified and then solved in the different wallet areas?

ulisesmac replied:
Regarding separating the flow, at the beginning it was my approach, but this infrastructure doesn't support it.

At the beginning, I was in a flow (the regular sending) and then just jumped to another flow when a collectible was picked (`wallet-flow` -> `send-collectible-flow`) and it worked for the simple sending, but once I wanted to skip some steps (particularly the collectible picking) then the flow didn't know it must jump to another flow.

So I realized that if I wanted that same screens take different flows and skip steps successfully, I should, from the beginning, stablish a flow and persist it on re-frame, and not passing the flow-id to the wizard anymore, UNLESS, you wanted to jump from one flow to another, and (IDK exactly how and when) make sure to clean the db with the flow picked, because the user could just give up, and close the flow even if it's not completed.


Since this required a significant change in the structure I thought it was not intended to do so, so I just kept our `wallet-flow` sharing all steps but skipping according to conditions based on the type of transaction.

I'm not against making big changes to the mechanism, I just was unsure, so if you think that is the best, I can do it 👍

Also, idk if we want multiple flows repeating many steps (as firsts and last steps in sending) or we want to define a flow with subflows (maybe a bit more complex to implement).

Wdyt?

J-Son89 commented on src/status_im/contexts/wallet/send/events.cljs:
```diff
@@ -99,27 +100,37 @@
+           [:dispatch
+            [:wallet/wizard-navigate-forward
+             {:current-screen current-screen
+              :flow-id        :wallet-flow}]]]})))
```
the idea here was to have the different flows marked.
i.e :asset-flow, :bridge-flow, :collectible-flow 👍 

I think we should use it this way and not mix the logic of these as they are all slightly different and it will be cleaner 👍

ulisesmac replied:
It was due to the infra not supporting it. Answered in https://github.com/status-im/status-mobile/pull/19515#discussion_r1551949724

ulisesmac commented:
> @ulisesmac - I tested out this branch. 
> It works great for sending an asset.
> One thought - there is also sending an asset from the home page by long press on a collectible.
> @mmilad75 has added the "Select From" page for this flow already so it should be rather straightforward to include that. 
> Shall we add it here too? 🤔 

Sure!, I'll add it :)

ulisesmac commented:
> > @ulisesmac - I tested out this branch.
> > It works great for sending an asset.
> > One thought - there is also sending an asset from the home page by long press on a collectible.
> > @mmilad75 has added the "Select From" page for this flow already so it should be rather straightforward to include that.
> > Shall we add it here too? 🤔
> 
> Sure!, I'll add it :)

I already added it, it wasn't as straightforward as I thought but it's not too complex, I added two comments in the code to make this more clear.

ulisesmac commented:
@OmarBasem @J-Son89 
Can you please review again?
additionally, I think QA can be skipped, wdyt?

J-Son89 commented:
> @OmarBasem @J-Son89 Can you please review again? additionally, I think QA can be skipped, wdyt?

I tend to disagree, I think this will benefit from QA. Send is quite an important area and we're making quite a number of changes which can break it, no?

briansztamfater commented on src/status_im/contexts/wallet/send/flow_config.cljs:
```diff
@@ -1,13 +1,30 @@
+   {:screen-id  :screen/wallet.select-collectible-amount
+    :skip-step? (fn [db]
+                  (or (not (collectible-selected? db))
+                      (some? (get-in db [:wallet :ui :send :amount]))))}
```
`amount` should be greater than 1, so probably we should skip if amount == 1 because user could only send 1  🤔

J-Son89 replied:
we have multiple collectibles now :)

ulisesmac replied:
this `amount` is not the owned collectible units, this `amont` key represents the desired number of collectibles to send and we are skipping this step if it already exists in the db.
i.e.
During this step we set the value in `[:wallet :ui :send :amount]`, if it has been already set, we skip it.

J-Son89 replied:
yes, but the user can decide how many at a time they want to send. 
see @vkjr's recent pr:
https://github.com/status-im/status-mobile/pull/19386

J-Son89 replied:
i,e https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=10771-86609&mode=design&t=XuAtw4Ji54tLkXJS-4

although, maybe we're talking about the same thing? 
🤔 
in any case this should only be for collectibles that are owned, so if this is a stretch in functionality, let's do a follow up instead.

briansztamfater replied:
> this `amount` is not the owned collectible units, this `amont` key represents the desired number of collectibles to send and we are skipping this step if it already exists in the db. i.e. During this step we set the value in `[:wallet :ui :send :amount]`, if it has been already set, we skip it.

I see, thanks for the explanation @ulisesmac.

> yes, but the user can decide how many at a time they want to send. see @vkjr's recent pr: #19386

@J-Son89 What I meant is that if user only owns 1 of the selected collectible, it doesn't make sense to show `:screen/wallet.select-collectible-amount`. But I see the case is already covered by this part
```
     {:db (cond-> collectible-tx
            one-collectible? (assoc-in [:wallet :ui :send :amount] 1))
```
in `:wallet/set-collectible-to-send` event, and the condition in skip-step?.

Looks good to me 👍

ulisesmac replied:
@J-Son89 
Yeah, that flow is being preserved, and it's only shown when the user has more than one collectible.

ok! @briansztamfater  :)

J-Son89 commented on src/status_im/contexts/wallet/collectible/view.cljs:
```diff
@@ -28,13 +29,17 @@
      :size            40
-     :icon-left       :i/send}
+     :icon-left       :i/send
+     :on-press        #(rf/dispatch
```
🙌 
(also this is one reason it's good to test. It's technically new functionality 👍

OmarBasem reviewed (approved):
Looks good, but would be better to add tests for the new/updated send events as it is a sensitive functionality and I think it broke a few times before

ulisesmac commented:
> Looks good, but would be better to add tests for the new/updated send events as it is a sensitive functionality and I think it broke a few times before

I considered it before, but I also thought these events modified are just adding like 1 or 2 keys, they mainly rely on the correctness of the wallet flow. But yeah, I can add some test just to check those keys exist.

status-im-auto commented:
## 88% of end-end tests have passed
```
Total executed tests: 52
Failed tests: 5
Expected to fail tests: 1
Passed tests: 46
```
```
IDs of failed tests: 702777,704613,703297,702786,702775 
```
```
IDs of expected to fail tests: 703503 
```
<h3>Failed tests (5)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&PR_ID=19515&TR_CASE_IDS=702777,704613,703297,702786,702775">Rerun failed tests</a></li><br/><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518715">test_group_chat_send_image_save_and_share</a>, id: 703297 </b></td></tr><tr><td><p><blockquote><div>Device 3: Find `ChatsTab` by `accessibility id`: `chats-stack-tab` </div><div>Device 3: Tap on found: ChatsTab </div></blockquote></p>

```
critical/chats/test_group_chat.py:283: in test_group_chat_send_image_save_and_share
    self.errors.verify_no_errors()
base_test_case.py:190: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Not expected image is shown to the admin.
E    Not expected image is shown to the member_2.
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7f101aee1a57498480822d1445b4f820?auth=a83161fb5ea4a5930cbde32443b828b2#265">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a0ecfb2268a9436f840809d9d465be82?auth=7b90172362d34a1e41efd0ed8ad4a842#177">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/583a5a60423c48119b0a97b3995d5f5e?auth=b72e0fb5c93635952e9a7ebee1ba398c#146">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityCenterContactRequestMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518683">test_add_contact_field_validation</a>, id: 702777 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `xpath`: `//*[@text="Paste"]` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
activity_center/test_activity_center.py:237: in test_add_contact_field_validation
    self.errors.verify_no_errors()
base_test_case.py:190: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Error message is not shown for value "zQ3shWGdfo6UvFLsX4RzkevoayZRyR2tzYqqPer44mUKLwgq"
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8ef29f8f524c4b1cacedd5db65d501b0?auth=475b09ed2c5bb7249538b00f15301c8a#134">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/65bbb8cba78349edb1413a1c28d4aa19?auth=f1bb0725b56f17cc080df5e1848b527a#196">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMergedTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518687">test_community_mentions_push_notification</a>, id: 702786 </b></td></tr><tr><td><p><blockquote><div>Device 2: Getting PN by 'user_2' </div><div>Device 2: Looking for a message by text: user_2 </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:931: in test_community_mentions_push_notification
    self.errors.verify_no_errors()
base_test_case.py:190: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Edited message is not shown correctly for the sender
E    Edited message is not shown correctly for the (receiver) admin
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/fa86edf7911746e7a0c29d36d7123c7b?auth=309eacbff45e1b24bda9d1f6536f32e5#135">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/97ed927cc6b94642afc7a802d3a13017?auth=ac789c10a7f77227bec42c55016bb17b#110">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestDeepLinksOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518722">test_links_open_universal_links_from_chat</a>, id: 704613 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `xpath`: `//*[@text="open community"]` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
critical/test_deep_and_universal_links.py:69: in test_links_open_universal_links_from_chat
    self.errors.verify_no_errors()
base_test_case.py:190: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Community 'Closed communtiy for e2e' was not requested to join by the url https://status.app/c/G0UAAGS9MeHbo_RoNMnOhOKTCo-fDDbgBCCyNNSN7qrzTqIgLem56TyzYYLf36askV7yGcros8muAdgB#zQ3shVWCcsH94vjf7WrFqBXx5MMAui2EPYMBme8ntACNWWVN3
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ffbb3e7799664b45b4e802b06ad34445?auth=3fc5bf07191ab4b032e50903cfb7ba73#53">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518682">test_links_deep_links</a>, id: 702775 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `BrowserTab` by `accessibility id`: `browser-stack-tab` </div><div>Device 1: Tap on found: BrowserTab </div></blockquote></p>

```
critical/test_deep_and_universal_links.py:112: in test_links_deep_links
    self.errors.verify_no_errors()
base_test_case.py:190: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Community 'Closed communtiy for e2e' was not requested to join by the deep link status.app://c/G0UAAGS9MeHbo_RoNMnOhOKTCo-fDDbgBCCyNNSN7qrzTqIgLem56TyzYYLf36askV7yGcros8muAdgB#zQ3shVWCcsH94vjf7WrFqBXx5MMAui2EPYMBme8ntACNWWVN3
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ffbb3e7799664b45b4e802b06ad34445?auth=3fc5bf07191ab4b032e50903cfb7ba73#192">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Expected to fail tests (1)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518720">test_community_discovery</a>, id: 703503 </b></td></tr><tr><td>

```
Test is not run, e2e blocker  
```

<b>[[reason: [NOTRUN] Curated communities not loading, https://github.com/status-im/status-mobile/issues/17852]]</b><br/><br/></td></tr></tbody></table></details><h3>Passed tests (46)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestActivityCenterContactRequestMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518701">test_activity_center_contact_request_accept_swipe_mark_all_as_read</a>, id: 702851 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8ef29f8f524c4b1cacedd5db65d501b0?auth=475b09ed2c5bb7249538b00f15301c8a#99">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/65bbb8cba78349edb1413a1c28d4aa19?auth=f1bb0725b56f17cc080df5e1848b527a#110">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518700">test_activity_center_contact_request_decline</a>, id: 702850 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8ef29f8f524c4b1cacedd5db65d501b0?auth=475b09ed2c5bb7249538b00f15301c8a#57">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/65bbb8cba78349edb1413a1c28d4aa19?auth=f1bb0725b56f17cc080df5e1848b527a#56">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518684">test_1_1_chat_emoji_send_reply_and_open_link</a>, id: 702782 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d1a73f53f69047049a0c3d07a0a34af2?auth=4a19d2a3c16a67e5995a3e00390b3fc2#140">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c44e961f0d294895a522fa5bfced9ed2?auth=a62d05db25e70757d5b55d3e91afa02b#120">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518679">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d1a73f53f69047049a0c3d07a0a34af2?auth=4a19d2a3c16a67e5995a3e00390b3fc2#546">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c44e961f0d294895a522fa5bfced9ed2?auth=a62d05db25e70757d5b55d3e91afa02b#397">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518691">test_1_1_chat_push_emoji</a>, id: 702813 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d1a73f53f69047049a0c3d07a0a34af2?auth=4a19d2a3c16a67e5995a3e00390b3fc2#402">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c44e961f0d294895a522fa5bfced9ed2?auth=a62d05db25e70757d5b55d3e91afa02b#259">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518681">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d1a73f53f69047049a0c3d07a0a34af2?auth=4a19d2a3c16a67e5995a3e00390b3fc2#340">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c44e961f0d294895a522fa5bfced9ed2?auth=a62d05db25e70757d5b55d3e91afa02b#205">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518702">test_1_1_chat_edit_message</a>, id: 702855 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d1a73f53f69047049a0c3d07a0a34af2?auth=4a19d2a3c16a67e5995a3e00390b3fc2#457">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c44e961f0d294895a522fa5bfced9ed2?auth=a62d05db25e70757d5b55d3e91afa02b#294">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518717">test_1_1_chat_send_image_save_and_share</a>, id: 703391 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d1a73f53f69047049a0c3d07a0a34af2?auth=4a19d2a3c16a67e5995a3e00390b3fc2#485">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c44e961f0d294895a522fa5bfced9ed2?auth=a62d05db25e70757d5b55d3e91afa02b#337">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518677">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d1a73f53f69047049a0c3d07a0a34af2?auth=4a19d2a3c16a67e5995a3e00390b3fc2#202">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c44e961f0d294895a522fa5bfced9ed2?auth=a62d05db25e70757d5b55d3e91afa02b#165">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518676">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d1a73f53f69047049a0c3d07a0a34af2?auth=4a19d2a3c16a67e5995a3e00390b3fc2#88">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c44e961f0d294895a522fa5bfced9ed2?auth=a62d05db25e70757d5b55d3e91afa02b#85">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUiTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518686">test_1_1_chat_delete_via_long_press_relogin</a>, id: 702784 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4db13963a3fd419987971807f610f78a?auth=d290c94b295fb4477275b00e5eeeec0d#175">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/098a4275bab64fa7a563dae554c7b058?auth=24d2173527c187f99843ebc79c2e6b92#126">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518685">test_1_1_chat_is_shown_message_sent_delivered_from_offline</a>, id: 702783 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4db13963a3fd419987971807f610f78a?auth=d290c94b295fb4477275b00e5eeeec0d#88">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/098a4275bab64fa7a563dae554c7b058?auth=24d2173527c187f99843ebc79c2e6b92#85">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518719">test_1_1_chat_mute_chat</a>, id: 703496 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4db13963a3fd419987971807f610f78a?auth=d290c94b295fb4477275b00e5eeeec0d#103">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/098a4275bab64fa7a563dae554c7b058?auth=24d2173527c187f99843ebc79c2e6b92#113">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMergedTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518690">test_community_markdown_support</a>, id: 702809 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/fa86edf7911746e7a0c29d36d7123c7b?auth=309eacbff45e1b24bda9d1f6536f32e5#191">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/97ed927cc6b94642afc7a802d3a13017?auth=ac789c10a7f77227bec42c55016bb17b#175">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518708">test_community_hashtag_links_to_community_channels</a>, id: 702948 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/fa86edf7911746e7a0c29d36d7123c7b?auth=309eacbff45e1b24bda9d1f6536f32e5#332">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/97ed927cc6b94642afc7a802d3a13017?auth=ac789c10a7f77227bec42c55016bb17b#330">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518698">test_community_leave</a>, id: 702845 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/fa86edf7911746e7a0c29d36d7123c7b?auth=309eacbff45e1b24bda9d1f6536f32e5#329">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/97ed927cc6b94642afc7a802d3a13017?auth=ac789c10a7f77227bec42c55016bb17b#299">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518721">test_community_join_when_node_owner_offline</a>, id: 703629 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/fa86edf7911746e7a0c29d36d7123c7b?auth=309eacbff45e1b24bda9d1f6536f32e5#539">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/97ed927cc6b94642afc7a802d3a13017?auth=ac789c10a7f77227bec42c55016bb17b#517">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518712">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ed9216b2c8544b04a1028de94dc5e859?auth=b7701039885009cdad3f5b45ada5dda1#250">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518680">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ed9216b2c8544b04a1028de94dc5e859?auth=b7701039885009cdad3f5b45ada5dda1#77">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518704">test_community_undo_delete_message</a>, id: 702869 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ed9216b2c8544b04a1028de94dc5e859?auth=b7701039885009cdad3f5b45ada5dda1#106">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518699">test_community_navigate_to_channel_when_relaunch</a>, id: 702846 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ed9216b2c8544b04a1028de94dc5e859?auth=b7701039885009cdad3f5b45ada5dda1#53">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518716">test_community_mute_community_and_channel</a>, id: 703382 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ed9216b2c8544b04a1028de94dc5e859?auth=b7701039885009cdad3f5b45ada5dda1#125">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePRTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518709">test_activity_center_mentions</a>, id: 702957 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9ea1ecb1252149eba9e69d0bcb00dc12?auth=f037964ac03b7f6e78bd87bf60662719#120">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b33f9f2e57a949148dcdd44db7c3a3bc?auth=5425f20b71db297d8bc6288bf871f7e7#102">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518710">test_activity_center_admin_notification_accept_swipe</a>, id: 702958 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9ea1ecb1252149eba9e69d0bcb00dc12?auth=f037964ac03b7f6e78bd87bf60662719#168">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b33f9f2e57a949148dcdd44db7c3a3bc?auth=5425f20b71db297d8bc6288bf871f7e7#118">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518678">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7f101aee1a57498480822d1445b4f820?auth=a83161fb5ea4a5930cbde32443b828b2#346">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a0ecfb2268a9436f840809d9d465be82?auth=7b90172362d34a1e41efd0ed8ad4a842#242">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/583a5a60423c48119b0a97b3995d5f5e?auth=b72e0fb5c93635952e9a7ebee1ba398c#239">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518718">test_group_chat_mute_chat</a>, id: 703495 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7f101aee1a57498480822d1445b4f820?auth=a83161fb5ea4a5930cbde32443b828b2#470">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a0ecfb2268a9436f840809d9d465be82?auth=7b90172362d34a1e41efd0ed8ad4a842#288">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/583a5a60423c48119b0a97b3995d5f5e?auth=b72e0fb5c93635952e9a7ebee1ba398c#257">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518714">test_group_chat_reactions</a>, id: 703202 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7f101aee1a57498480822d1445b4f820?auth=a83161fb5ea4a5930cbde32443b828b2#160">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a0ecfb2268a9436f840809d9d465be82?auth=7b90172362d34a1e41efd0ed8ad4a842#107">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/583a5a60423c48119b0a97b3995d5f5e?auth=b72e0fb5c93635952e9a7ebee1ba398c#87">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518688">test_group_chat_join_send_text_messages_push</a>, id: 702807 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7f101aee1a57498480822d1445b4f820?auth=a83161fb5ea4a5930cbde32443b828b2#141">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a0ecfb2268a9436f840809d9d465be82?auth=7b90172362d34a1e41efd0ed8ad4a842#79">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/583a5a60423c48119b0a97b3995d5f5e?auth=b72e0fb5c93635952e9a7ebee1ba398c#79">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518689">test_group_chat_offline_pn</a>, id: 702808 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7f101aee1a57498480822d1445b4f820?auth=a83161fb5ea4a5930cbde32443b828b2#309">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a0ecfb2268a9436f840809d9d465be82?auth=7b90172362d34a1e41efd0ed8ad4a842#216">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/583a5a60423c48119b0a97b3995d5f5e?auth=b72e0fb5c93635952e9a7ebee1ba398c#205">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518725">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5a79e3283f384de589303dae4f45b249?auth=ace0bef8645b0236627351b17cdae8dc#113">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5b69d07bd74f4c5cb61a1bbc7741319f?auth=476892d9c0fdd7a5f112bef2c6ceaa65#92">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518724">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5a79e3283f384de589303dae4f45b249?auth=ace0bef8645b0236627351b17cdae8dc#40">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5b69d07bd74f4c5cb61a1bbc7741319f?auth=476892d9c0fdd7a5f112bef2c6ceaa65#40">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518706">test_navigation_jump_to</a>, id: 702936 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4883458838554e11b3ba542908663a50?auth=b31bcf65a577a486aa0b312ba9495b1d#140">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/76b032c1254149e38d7e314f85d6d55b?auth=3ff50ea9e3c06294cd01ba1a30fe878c#120">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518707">test_activity_center_reply_read_unread_delete_filter_swipe</a>, id: 702947 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4883458838554e11b3ba542908663a50?auth=b31bcf65a577a486aa0b312ba9495b1d#168">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/76b032c1254149e38d7e314f85d6d55b?auth=3ff50ea9e3c06294cd01ba1a30fe878c#123">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518726">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4761e0c224b04139af1756cb8bc6434b?auth=4ead839a5199c857ec13ec5bfa46840d#39">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518727">test_wallet_add_remove_watch_only_account</a>, id: 727232 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4761e0c224b04139af1756cb8bc6434b?auth=4ead839a5199c857ec13ec5bfa46840d#103">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518713">test_community_several_images_send_reply</a>, id: 703194 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d971992412aa4a36bc393ee49a3603c0?auth=8f819f35a104e8d64a8715a0dce7f4d9#213">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d2a9e7fdc964462f96fd7fa3281aa7a8?auth=441b30480f698e68834a9275fa6a1a7e#165">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518703">test_community_one_image_send_reply</a>, id: 702859 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d971992412aa4a36bc393ee49a3603c0?auth=8f819f35a104e8d64a8715a0dce7f4d9#238">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d2a9e7fdc964462f96fd7fa3281aa7a8?auth=441b30480f698e68834a9275fa6a1a7e#229">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518694">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d971992412aa4a36bc393ee49a3603c0?auth=8f819f35a104e8d64a8715a0dce7f4d9#286">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d2a9e7fdc964462f96fd7fa3281aa7a8?auth=441b30480f698e68834a9275fa6a1a7e#285">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518711">test_community_mark_all_messages_as_read</a>, id: 703086 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d971992412aa4a36bc393ee49a3603c0?auth=8f819f35a104e8d64a8715a0dce7f4d9#547">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d2a9e7fdc964462f96fd7fa3281aa7a8?auth=441b30480f698e68834a9275fa6a1a7e#428">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518705">test_community_contact_block_unblock_offline</a>, id: 702894 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d971992412aa4a36bc393ee49a3603c0?auth=8f819f35a104e8d64a8715a0dce7f4d9#392">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d2a9e7fdc964462f96fd7fa3281aa7a8?auth=441b30480f698e68834a9275fa6a1a7e#351">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518723">test_community_edit_delete_message_when_offline</a>, id: 704615 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d971992412aa4a36bc393ee49a3603c0?auth=8f819f35a104e8d64a8715a0dce7f4d9#586">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d2a9e7fdc964462f96fd7fa3281aa7a8?auth=441b30480f698e68834a9275fa6a1a7e#458">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518693">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d971992412aa4a36bc393ee49a3603c0?auth=8f819f35a104e8d64a8715a0dce7f4d9#196">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d2a9e7fdc964462f96fd7fa3281aa7a8?auth=441b30480f698e68834a9275fa6a1a7e#136">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518692">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d971992412aa4a36bc393ee49a3603c0?auth=8f819f35a104e8d64a8715a0dce7f4d9#154">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d2a9e7fdc964462f96fd7fa3281aa7a8?auth=441b30480f698e68834a9275fa6a1a7e#107">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>9. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518697">test_community_links_with_previews_github_youtube_twitter_gif_send_enable</a>, id: 702844 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d971992412aa4a36bc393ee49a3603c0?auth=8f819f35a104e8d64a8715a0dce7f4d9#310">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d2a9e7fdc964462f96fd7fa3281aa7a8?auth=441b30480f698e68834a9275fa6a1a7e#300">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>10. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518696">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d971992412aa4a36bc393ee49a3603c0?auth=8f819f35a104e8d64a8715a0dce7f4d9#172">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d2a9e7fdc964462f96fd7fa3281aa7a8?auth=441b30480f698e68834a9275fa6a1a7e#124">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>11. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518695">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d971992412aa4a36bc393ee49a3603c0?auth=8f819f35a104e8d64a8715a0dce7f4d9#355">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d2a9e7fdc964462f96fd7fa3281aa7a8?auth=441b30480f698e68834a9275fa6a1a7e#341">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

VolodLytvynenko commented:
Hi @ulisesmac . Could you please resolve existing conflicts and rebase PR? thanx!

ulisesmac commented:
> Hi @ulisesmac . Could you please resolve existing conflicts and rebase PR? thanx!

@VolodLytvynenko Done!

VolodLytvynenko commented:
Hi @ulisesmac Thank you for PR. Take a look at the found issue:

## PR_ISSUE 1: The assets can't be sent on 'send to' page when an asset is selected on account overview page

### Steps:
1. Restore the user with available assets (example: eth)
2. Go to the account overview page
3. Long tap on the asset -> send
4. Select account with available assets to send from (eth)
5. Navigate to sent to page

### Actual result:
User sees 0 assets on the 'send to' page.

https://github.com/status-im/status-mobile/assets/52490791/27f047b6-a356-4fcc-9638-925f31cbe5ff

### Expected result:
User sees available assets on the 'sent to' page

### OS: 
IOS, Android

### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17

VolodLytvynenko commented:
I'm not sure if this issue is within the scope of this PR. I think it would be better to implement the collectible detail page for watch-only collectibles before this PR. This is because currently, this PR allows going to the send flow for watch-only collectibles. 

However, this could be considered as a follow-up. Is it possible to implement this page within the current PR, or would it be better to do it separately? WDYT?

### Step:
1. Add watch only account with some collectibles (example: `0x8d2413447ff297d30bdc475f6d5cb00254685aae`. Take a look some collectibles are not fetched until relogin due existing issue https://github.com/status-im/status-mobile/issues/18613 )
2. Open watch only collectible

### Actual result:
'Send' button is shown for watch-only collectibles and can navigate to the 'send' flow
![image](https://github.com/status-im/status-mobile/assets/52490791/a2a8d70d-8789-444c-83f9-7ff7db467c83)



### Expected result:
Watch-only collectible page is shown as per design without 'send' button
![image](https://github.com/status-im/status-mobile/assets/52490791/cda7c74a-4a86-45ff-89d6-7992f2762fac)

https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=12077-156467&mode=design&t=Qzwvre6uKEljOmd6-0
### OS: 
IOS, Android

### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17

VolodLytvynenko commented:
## PR_ISSUE 3: The select assets screen is shown when a watch-only token is going to be sent

### Steps to reproduce:
1. Add watch only an account with some assets that are not available on the regular accounts (example: `0x8d2413447ff297d30bdc475f6d5cb00254685aae`)
2. Long tap on the asset that is available only within the current watch-only account
3. Select an address to 'send to'
4. Check next page


### Actual result:
The 'Select assets' page is shown, even though the asset has already been selected in step 2.

https://github.com/status-im/status-mobile/assets/52490791/51afec6b-0175-4a6f-af67-71c0e62af1ff

### Expected result:
Potential solutions:
- The 'Select assets' page should not be shown since the asset has already been selected.
- probably if the asset is related only to a watch-only account, the drawer should not be shown. Let's wait for the design team's response https://discord.com/channels/624634427930312714/852533718790570015/1228346181097230539

### OS: 
IOS, Android
### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17
### Logs

VolodLytvynenko commented:
## PR_ISSUE 4: The 'From' page is shown after sending the collectible from the account overview page

### Steps:
1. Restore the account with collectibles
2. Long tap collectible on the account overview page
3. Tap 'send'

### Actual result:
The 'from' screen is shown

https://github.com/status-im/status-mobile/assets/52490791/4bf7b0a8-8b8a-4fc2-8f2e-d99c253b7fa4

### Expected result:
the 'from' screen is missed for collectibles sending flow


### OS: 
IOS, Android

### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17

VolodLytvynenko commented:
## PR_ISSUE 5: The routes are not found when attempting to send a collectible using the 'send' button in the opened collectible

### Steps:
1. Restore the account with collectibles
2. Long tap collectible
3. Tap 'send'

### Actual result:
The routes are not found

https://github.com/status-im/status-mobile/assets/52490791/d90ccee5-ae90-4f86-879f-91a1e793e0a3

### Expected result:
The routes are found

### Logs:
[Status-debug-logs (8).zip](https://github.com/status-im/status-mobile/files/14960326/Status-debug-logs.8.zip)

ulisesmac commented:
Hey @VolodLytvynenko 
sorry for the late response.

### Issue 1

This is due to the relogin issue, so I'd say it's out of scope of this one

### Issue 2 and 3

This PR is not focused on fixing UI things, so they need to be addressed but preferably in a separate one.

### Issue 4
This issue is also related to the recovery and relogin one, so I think we should address it separately.

### Issue 5

We don't know the exact reason for routes not being found, but it's out of scope of this PR.


So, this PR is about connecting some flows, but they may still be buggy for various reasons, we'll address them separately.

VolodLytvynenko commented:
Hi @ulisesmac thank you for clarification. PR is ready to be merged

VolodLytvynenko commented:
@ulisesmac please let me know if any additional tests are needed after resolving the conflicts in this PR. I can quickly check everything

ulisesmac committed:
Fix typo `:collecible` -> `:collectible`

ulisesmac committed:
Extend the wallet-send-flow for collectibles and tokens in all entry points

ulisesmac committed:
Lint fix

ulisesmac commented:
@VolodLytvynenko I solved the conflicts  and merged the PR, I'll open issues for the points you pointed out and I'll be opening PRs to address them. Thank you!

ulisesmac commented:
> ## PR_ISSUE 3: The select assets screen is shown when a watch-only token is going to be sent
> ### Steps to reproduce:
> 
>     1. Add watch only an account with some assets that are not available on the regular accounts (example: `0x8d2413447ff297d30bdc475f6d5cb00254685aae`)
> 
>     2. Long tap on the asset that is available only within the current watch-only account
> 
>     3. Select an address to 'send to'
> 
>     4. Check next page
> 
> 
> ### Actual result:
> 
> The 'Select assets' page is shown, even though the asset has already been selected in step 2.
> select_as_page.mp4
> ### Expected result:
> 
> Potential solutions:
> 
>     * The 'Select assets' page should not be shown since the asset has already been selected.
> 
>     * probably if the asset is related only to a watch-only account, the drawer should not be shown. Let's wait for the design team's response https://discord.com/channels/624634427930312714/852533718790570015/1228346181097230539
> 
> 
> ### OS:
> 
> IOS, Android
> ### Devices:
> 
>     * Pixel 7a, Android 13
> 
>     * iPhone 11 Pro Max, IOS 17


An update @VolodLytvynenko 

This is being addressed by @mmilad75  in:
- https://github.com/status-im/status-mobile/pull/20050

ulisesmac commented:
@VolodLytvynenko Another update on the issues, I've tested them on latest develop:

> PR_ISSUE 1: The assets can't be sent on 'send to' page when an asset is selected on account overview page

No longer reproducible.

> PR_ISSUE 2: 'Send' button is shown for watch-only collectibles and can navigate to the 'send' flow

No longer reproducible.

> PR_ISSUE 3: The select assets screen is shown when a watch-only token is going to be sent

As said above @mmilad75  is solving it.

> PR_ISSUE 4: The 'From' page is shown after sending the collectible from the account overview page

No longer reproducible.

> PR_ISSUE 5: The routes are not found when attempting to send a collectible using the 'send' button in the opened collectible

No longer reproducible. Maybe it was happening because the collectible didn't support sending, you didn't have enough eth or it was a side issue and now it's solved.


So there's nothing else pending from this PR, thank you for taking the time to test!

VolodLytvynenko commented:
> The select assets screen is shown when token is going to be sent which available on both watch-only and regular account #19745

Hi @ulisesmac thank you for the update. I confirm this. Issues you mentioned are fixed
